client/core/trade.go

```diff
@@ -628,6 +630,34 @@ func (t *trackedTrade) activeMatches() []*matchTracker {
 	return actives
 }
 
+// unspentContractAmounts returns the total amount locked in unspent swaps.
+func (t *trackedTrade) unspentContractAmounts(assetID uint32) (amount uint64) {
+	if t.wallets.fromAsset.ID != assetID {
+		// Only swaps sent from the specified assetID should count.
+		return 0
+	}
+	t.mtx.RLock()
+	defer t.mtx.RUnlock()
+	for _, match := range t.matches {
+		side, status := match.Match.Side, match.Match.Status
+		if status >= order.MakerRedeemed || len(match.MetaData.Proof.RefundCoin) != 0 {
+			// Any redemption or own refund implies our swap is spent.
+			// Even if we're Maker and our swap has not been redeemed
+			// by Taker, we should consider it spent.
+			continue
+		}
+		if (side == order.Maker && status >= order.MakerSwapCast) ||
+			(side == order.Taker && status == order.TakerSwapCast) {
+			swapAmount := match.Match.Quantity
+			if t.Trade().Sell {
+				swapAmount = calc.BaseToQuote(match.Match.Rate, match.Match.Quantity)
+			}
```

Comment:
This seems to be backwards.  Consider a maker buying 20 DCR @ 0.1.  The maker broadcasts a BTC contract for 2 BTC,  but `unspentContractAmounts(0)` for this trade computes `swapAmount` of 20 because it is a buy (buying DCR).  Seems like it needs to be checking `!Sell` and doing `QuoteToBase` instead of `BaseToQuote`.

Reply:
e.g. 

![image](https://user-images.githubusercontent.com/9373513/93111715-9b4b4380-f67c-11ea-9ba8-daec86e53c6c.png)

That should be showing 2 BTC locked and 20 DCR locked, not the other way around.


![image](https://user-images.githubusercontent.com/9373513/93111733-a43c1500-f67c-11ea-8cd1-57977bb8b9c2.png)

Reply:
BTW, I'm doing these tests with the "beta" dcr wallet to avoid confusion with locked stake outputs that are present in the alpha wallet.

Reply:
Hmm, seen. Should be `!Sell` but `BaseToQuote`. I'd referred
https://github.com/decred/dcrdex/blob/5f1b1691325511913ec21a6dc164481a9b2a54ee/client/core/trade.go#L804-L808
and
https://github.com/decred/dcrdex/blob/5f1b1691325511913ec21a6dc164481a9b2a54ee/client/core/trade.go#L835-L839

Missed the `!Sell`.

The fromAsset is the base asset for Sell and the quote asset for Buy. Since we're only calculating `unspentContractAmounts` for the fromAssetID, we can simply check if it's buy or sell. But I do think comparing the fromAssetID to the mkt's quote or base asset ID lends better to intuitive understanding.

Reply:
Checking against `t.Quote()` resolved it in https://github.com/decred/dcrdex/pull/683/commits/b13f12f85b26ddec9acbedc1965c594589ff9c96.

![image](https://user-images.githubusercontent.com/9373513/93115868-17945580-f682-11ea-8476-8689fd028e2d.png)

Locked-by-contract balances are correct now.